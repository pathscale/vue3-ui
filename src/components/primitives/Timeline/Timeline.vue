<script>
export default {
  name: 'Timeline',
  props: {
    stages: { type: Object, required: true },
  },
}
</script>

<template>
  <ol class="timeline">
    <li
      v-for="(stage, index) in stages"
      :key="index"
      :class="{
        'has-text-success is-active': stage.active,
        'has-text-grey': !stage.active,
        'has-text-danger': stage.error,
      }">
      <div>
        <slot :stage="stage" />
      </div>
    </li>
  </ol>
</template>
