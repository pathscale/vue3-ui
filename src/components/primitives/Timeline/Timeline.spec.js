import { shallowMount } from '@vue/test-utils'
import Timeline from './Timeline.vue'

let wrapper

describe('Timeline', () => {
  beforeEach(() => {
    wrapper = shallowMount(Timeline)
  })

  it('is called', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
