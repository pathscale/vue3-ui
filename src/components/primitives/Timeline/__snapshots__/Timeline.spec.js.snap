// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline render correctly 1`] = `<ol class="timeline"></ol>`;
